corehq/apps/accounting/forms.py

```diff
@@ -160,6 +160,18 @@ def __init__(self, account, *args, **kwargs):
             )
         )
 
+    def clean_name(self):
+        existing_names = set(account.name for account in BillingAccount.objects.all())
```

Comment:
Or even 
```
if BillingAccount.objects.filter(name = cleaned_data['name']).exists() and self.account and (self.account.name != cleaned_data['name'])
  raise ValidationError("Name '%s' is already taken." % name)
```
might be even more efficient given there's an index on `name`. `all()` could return a lot of objects? but this operation might not be too frequent